packages/api/src/routes/medical/__tests__/mapi/e2e/mapi.test.e2e.ts

```diff
@@ -0,0 +1,173 @@
+
+jest.setTimeout(30000);
+
+// NEVER TO BE RUN IN PRODUCTION
+if (Config.isStaging() || !Config.isCloudEnv()) {
+  describe("MAPI E2E Tests", () => {
+    let facility: Facility;
+    let patient: Patient;
+
+    it("gets an organization", async () => {
+      const org = await medicalApi.getOrganization();
+
+      if (!org) {
+        throw new Error("Org not found");
+      }
+
+      const fhirOrg = await fhirApi.readResource(ResourceType.Organization, org.id, fhirHeaders);
+
+      const cwOrg = await retryFunction<CWOrganization | undefined>(
+        async () => await cwCommands.organization.getOne(org.oid),
+        maxRetries
+      );
+
+      validateLocalOrg(org);
+      validateFhirOrg(fhirOrg, org);
+      validateCWOrg(cwOrg, org);
+    });
+
+    it("updates an organization", async () => {
+      const org = await medicalApi.getOrganization();
+
+      if (!org) {
+        throw new Error("Org not found");
+      }
+
+      const newName = faker.word.noun();
+
+      const updateOrg: Organization = {
+        ...org,
+        name: newName,
+      };
+
+      const updatedOrg = await medicalApi.updateOrganization(updateOrg);
+
+      await fhirApi.invalidateAll();
+      const fhirOrg: FhirOrg = await fhirApi.readResource(
+        ResourceType.Organization,
+        updatedOrg.id,
+        fhirHeaders
+      );
+
+      const cwOrg: CWOrganization | undefined = await retryFunction<CWOrganization | undefined>(
+        async () => await cwCommands.organization.getOne(updatedOrg.oid),
+        maxRetries,
+        result => result?.name === newName
+      );
+
+      validateLocalOrg(updatedOrg, updateOrg);
+      validateFhirOrg(fhirOrg, updateOrg);
+      validateCWOrg(cwOrg, updateOrg);
+    });
+
+    it("create a facility", async () => {
+      facility = await medicalApi.createFacility(createFacility);
+
+      validateFacility(facility);
+    });
+
+    it("updates a facility", async () => {
+      const newName = faker.word.noun();
+
+      const updateFacility: Facility = {
+        ...facility,
+        name: newName,
+      };
+
+      facility = await medicalApi.updateFacility(updateFacility);
+
+      validateFacility(facility, updateFacility);
+      expect(facility.name).toBe(newName);
+    });
+
+    it("gets one facility", async () => {
+      const getFacility = await medicalApi.getFacility(facility.id);
+
+      validateFacility(getFacility);
+    });
+
+    it("creates a patient", async () => {
+      const org = await medicalApi.getOrganization();
+
+      if (!org) {
+        throw new Error("Org not found");
+      }
+
+      patient = await medicalApi.createPatient(createPatient, facility.id);
+
+      const fhirPatient = await fhirApi.readResource(ResourceType.Patient, patient.id, fhirHeaders);
+
+      validateLocalPatient(patient);
+      validateFhirPatient(fhirPatient);
+
+      // Creating a CW patient is done in the background need to await so we can query docs
+      await Util.sleep(10000);
```

Comment:
I understand the need, but this is a long delay... I'm wondering if we could have a check w/ retries and a small delay between them (like 500ms). If the patient is available before, we don't need to wait 10s, also if its not ready in 10s we would not fail (could have a combination of retries/time that would allow for a long time, with a limit).

Also for other delays/sleeps.

Reply:
This issue is we cant query the patient in CW without getting the external data from a patient and we can get the external data from a patient without creating an internal route to expose that data. So it doesnt seem like there is a way really to validate the patient exists in cw

Reply:
Oh, I see what you mean. It makes sense.